File: brainTeaser.js
function brainTeaser(word){
  var x = word.split("")
  for (var i in x){
    j = i
    for(;j<x.length;j++){
      var stringToPrint = ""
      for( var k = i; k <= j; k++){
        stringToPrint += word[k]
      }
      console.log(stringToPrint + ", ")
    }
  }
}

brainTeaser("dog")

File: brain_teaser.rb
def brain_teaser()
  print "Please input a word, or type Q to quit"
  word = gets.strip
  if word == 'Q'
    exit
  end
  a = 0
  b = 0
  x = word.split("")
  while a < x.length
    b = a
    while b < x.length
      c = x.join("")
      print (c[a..b])
      print (", ")
      b += 1
    end
    a += 1
  end
  x = x.reverse
  a = 0
  b = 0
  while a < x.length
    b = a
    while b < x.length
      c = x.join("")
      print (c[a..b])
      print (", ")
      b += 1
    end
    a += 1
  end
end


brain_teaser()